platform/osx/os_osx.mm

```diff
@@ -1807,10 +1806,14 @@ static int get_screen_index(NSScreen *screen) {
 		CGFloat menuBarHeight = [[[NSApplication sharedApplication] mainMenu] menuBarHeight];
 		if (menuBarHeight != 0.f) {
 			size.y += menuBarHeight;
-#if MAC_OS_X_VERSION_MAX_ALLOWED <= 101104
 		} else {
-			size.y += [[NSStatusBar systemStatusBar] thickness];
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 101200
```

Comment:
Wouldn't a direct test without `#if` be equivalent and simpler?
```
if (floor(NSAppKitVersionNumber) < NSAppKitVersionNumber10_12) {
    size.y += [[NSStatusBar systemStatusBar] thickness];
}
```

Reply:
`#if MAC_OS_X_VERSION_MAX_ALLOWED` is for compiling on pre 10.12 since NSAppKitVersionNumber10_12 is not defined.

Reply:
Isn't it overkill to do both compile and runtime checks? For example currently if you build for 10.9 on 10.12, you will get a logger (albeit without the 10.12-specific part of it), but if you build for 10.9 on 10.9 you won't have it.

I think runtime checks might be enough, with custom defines of `NSAppKitVersionNumber10_{10,11,12}` when undefined, no?

Reply:
Though I guess in the case of the logger without the compiler defines it might fail build if older SDKs don't know the methods.

Reply:
Yep, without compile time checks it won't build without 10.12 SDK (min. Xcode 8.0 on macOS 10.11.5).